code/datums/mob_hud.dm

```diff
@@ -268,15 +268,21 @@ var/datum/mob_hud/huds = list(
 		var/datum/limb/head = get_limb("head")
 		var/datum/internal_organ/heart/heart = internal_organs_by_name["heart"]
 		var/revive_enabled = 1
+		var/stage = 1
 		if(world.time - timeofdeath > revive_grace_period || undefibbable)
 			revive_enabled = 0
-		else
-			if(suiciding || !head || !head.is_usable() || !heart || heart.is_broken() || !has_brain() || chestburst || (HUSK in mutations) || !mind)
+		else if(suiciding || !head || !head.is_usable() || !heart || heart.is_broken() || !has_brain() || chestburst || (HUSK in mutations) || !mind)
+			revive_enabled = 0
+		else if(!client)
+			var/mob/dead/observer/G = get_ghost()
+			if(!istype(G))
 				revive_enabled = 0
-			else if(!client)
-				var/mob/dead/observer/G = get_ghost()
-				if(!istype(G))
-					revive_enabled = 0
+
+		if(stat == DEAD && !undefibbable)
+			if((world.time - timeofdeath) > ((revive_grace_period / 4) * 2))
```

Comment:
```suggestion
			if((world.time - timeofdeath) > (revive_grace_period * 0.5))
```
Or `/ 2`. It's a little more clear than that double operation, though I see what you were doing.
`revive_grace_period` could be a constant instead, but it may be out of the scope of your PR. It's only used in 7 files.

Reply:
as per code standards avoid division.